core/ir/opnd_shared.c

```diff
@@ -1997,7 +1997,8 @@ reg_get_value_ex(reg_id_t reg, dr_mcontext_t *mc, OUT byte *val)
         *(reg_t *)val = regval;
     }
 #else
-    CLIENT_ASSERT(false, "NYI i#1551");
+    reg_t regval = reg_get_value(reg, mc);
```

Comment:
With some small refactoring of the function, you can share this code with those between lines 1996-1997.